lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -36,6 +36,33 @@ using namespace circt::om;
 
 namespace {
 
+static bool shouldCreateClassImpl(FModuleLike moduleLike) {
+  if (isa<firrtl::ClassLike>(moduleLike.getOperation()))
+    return true;
+
+  // Always create a class for public modules.
+  if (moduleLike.isPublic())
+    return true;
+
+  // Create a class for modules with property ports.
+  bool hasClassPorts = llvm::any_of(moduleLike.getPorts(), [](PortInfo port) {
+    return isa<PropertyType>(port.type);
+  });
+
+  if (hasClassPorts)
+    return true;
+
+  // Create a class for modules that instantiate classes or modules with
+  // property ports.
+  for (auto op :
+       moduleLike.getOperation()->getRegion(0).getOps<FInstanceLike>())
```

Comment:
Looks like the InstanceGraph is available at this point -- if so, instead of walking to gather instances, ask the instance graph?

Reply:
Yeah, thanks!